aep/general/0203/aep.md.j2

```diff
@@ -1,5 +1,278 @@
+
+A field **may** be described as optional if it is a field on a request message
+(a message that is an argument to an RPC, usually ending in `Request`), or a
+field on a submessage.
+
+### Output only
+
+The use of `OUTPUT_ONLY` indicates that the field is provided in responses, but
+that including the field in a message in a request does nothing (the server
+**must** clear out any value in this field and **must not** throw an error as a
+result of the presence of a value in this field on input). Similarly, services
+**must** ignore the presence of output only fields in update field masks (see:
+AIP-161).
+
+Additionally, a field **should** only be described as output only if it is a
+field in a resource message, or a field of a message farther down the tree.
+Notably, fields in response messages (a message which only ever acts as a
+return value to an RPC, usually ending in `Response`) **should not** be
+described as output only because this is already implied.
+
+Output only fields **may** be set to empty values if appropriate to the API.
+
+Potential use cases for output only fields (this is not an exhaustive list)
+are:
+
+- Create or update timestamps.
+- Derived or structured information based on original user input.
+- Properties of a resource assigned by the service which can not be altered.
+
+### Required
+
+The use of `REQUIRED` indicates that the field **must** be present (and set to
+a non-empty value) on the request or resource.
+
+A field **should** only be described as required if _either_:
+
+- It is a field on a resource that a user provides somewhere as input. In this
+  case, the resource is only valid if a "truthy" value is _stored_.
+  - When [creating](./standard-methods) the resource, a value **must** be
+    provided for the field on the create request.
+  - When [updating](./standard-methods) the resource, the user **may** omit the
+    field provided that the field is also absent from the field mask,
+    indicating no change to the field (otherwise it **must** be provided).
+- It is a field on a request message (a message that is an argument to an RPC,
+  with a name usually ending in `Request`). In this case, a value **must** be
+  provided as part of the request, and failure to do so **must** cause an error
+  (usually `INVALID_ARGUMENT`).
+
+We define the term "truthy" above as follows:
+
+- For primitives, values other than `0`, `0.0`, empty string/bytes, and `false`
+- For repeated fields maps, values with at least one entry
+- For messages, any message with at least one "truthy" field.
+
+Fields **should not** be described as required in order to signify:
+
+- A field which will always be present in a response.
+- A field which is conditionally required in some situations.
+- A field on any message (including messages that are resources) which is never
+  used as user input.
+
+**Note:** In most cases, empty values (such as `false` for booleans, `0` for
```

Comment:
this note further emphasizes to me that we need to remove this part of the guidance.

Why have a required boolean, if the value for a required boolean must always be true?

proto3 the optional keyword to differentiate between unset and set values.

Reply:
If we're only supporting proto3, I agree with this.

Reply:
In proto2 - what value would a boolean that is always set to true serve? Would the idea be that it's a flag that helps others understand that a new version of the schema is used?

Reply:
I'm re-reading this again and you're right. My brain was substituting "must always be true" for "defaults to true"